rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/CoordinatorBookKeeper.java

```diff
@@ -43,24 +54,91 @@
   protected Cache<String, Boolean> fileValidatedWorkerCache;
 
   private final boolean isValidationEnabled;
+  private static Integer lock = 1;
+  protected ClusterManager clusterManager;
+  int clusterType;
 
-  public CoordinatorBookKeeper(Configuration conf, MetricRegistry metrics) throws FileNotFoundException
+  public CoordinatorBookKeeper(Configuration conf, MetricRegistry metrics) throws BookKeeperInitializationException
   {
     this(conf, metrics, Ticker.systemTicker());
   }
 
   @VisibleForTesting
-  public CoordinatorBookKeeper(Configuration conf, MetricRegistry metrics, Ticker ticker) throws FileNotFoundException
+  protected CoordinatorBookKeeper(Configuration conf, MetricRegistry metrics, Ticker ticker) throws BookKeeperInitializationException
   {
     super(conf, metrics, ticker);
     this.isValidationEnabled = CacheConfig.isValidationEnabled(conf);
     this.liveWorkerCache = createHealthCache(conf, ticker);
     this.cachingValidatedWorkerCache = createHealthCache(conf, ticker);
     this.fileValidatedWorkerCache = createHealthCache(conf, ticker);
+    this.clusterType = CacheConfig.getClusterType(conf);
```

Comment:
default cluster type in configs is TEST_CLUSTER_MANAGER ... This change will need a change in all deployments of rubix to set the configs right, any scripts setting up rubix will need to be changed.

Reply:
The script that starts the rubix daemons will pass the cluster type. So for unit tests, it will be set to TEST_CLUSTER_MANAGER, this will be overridden with the incoming config